opm/parser/eclipse/EclipseState/Schedule/Group/Group.hpp

```diff
@@ -246,6 +246,7 @@ struct ProductionControls {
     const GroupProductionProperties& productionProperties() const;
     const std::map<Phase , GroupInjectionProperties>& injectionProperties() const;
     const GroupType& getGroupType() const;
+    const UnitSystem& getUnitSystem() const;
```

Comment:
That the `Group` object has access to a  `UnitSystem`  is more of an implementation accident - we should not cement this by using `Group` class a source of a `UnitSystem`. My suggested alternative is to pass it in as an argument to the `write_WELSPECS` function. At the callsite in `EclipseIO` the unit system can be had from:
```
const auto& unit_system = this->impl->es.getUnits();
```